pkg/policy/api/selector.go

```diff
@@ -254,7 +277,12 @@ func (n *EndpointSelector) Matches(lblsToMatch k8sLbls.Labels) bool {
 		}
 	}
 
-	return lbSelector.Matches(lblsToMatch)
+	for _, req := range *n.requirements {
```

Comment:
If you are going to rely on n.requirements. How can you be sure the `LabelSelector` didn't change between `Matches`?

Reply:
The `EndpointSelector` shouldn't be modified except by using the methods defined within this file such as `AddMatch()`. Perhaps we could consider hiding the LabelSelector to enforce this.